controllers/rcj_soccer_referee_supervisor/referee/supervisor.py

```diff
@@ -111,60 +118,99 @@ def _update_positions(self):
             # TODO: update self.robot_in_penalty_counter if the robot
             #       is located in penalty area
 
-    def draw_scores(self, blue: int, yellow: int):
+    def add_event_subscriber(self, subscriber: EventHandler):
+        """Add new event subscriber.
+
+        Args:
+            subscriber (EventHandler): Instance inheriting EventHandler
         """
-        Visualize (draw) the provide scores for both the blue and the yellow
-        teams.
+        self.eventer.subscribe(subscriber)
+
+    def draw_scores(self, blue: int, yellow: int):
+        """Visualize (draw) the provide scores for both the blue and
+        the yellow teams.
 
         Args:
             blue (int): score of the blue team
             yellow (int): score of the yellow team
         """
 
-        self.setLabel(0, str(blue),
-                      # X and Y positions
-                      0.92, 0.01,
-                      # Size and color
-                      0.1, 0x0000ff,
-                      # Transparency and Font
-                      0.0, "Tahoma")
-
-        self.setLabel(1, str(yellow),
-                      # X and Y positions
-                      0.05, 0.01,
-                      # Size and color
-                      0.1, 0xffff00,
-                      # Transparency and Font
-                      0.0, "Tahoma")
-
-    def draw_time(self, time):
-        """
-        Visualize (draw) the current match time
+        self.setLabel(
+            LabelIds.BLUE_SCORE,
+            str(blue),
+            0.92,  # X position
+            0.01,  # Y position
+            0.1,  # Size
+            0x0000ff,  # Color
+            0.0,  # Transparency
+            "Tahoma",  # Font
+        )
+
+        self.setLabel(
+            LabelIds.YELLOW_SCORE,
+            str(yellow),
+            0.05,  # X position
+            0.01,  # Y position
+            0.1,  # Size
+            0xffff00,  # Color
+            0.0,  # Transparency
+            "Tahoma"  # Font
+        )
+
+    def draw_time(self, time: int):
+        """Visualize (draw) the current match time
 
         Args:
             time (int): the current match time
         """
 
-        time_str = "%02d:%02d" % (time // 60, time % 60)
-        self.setLabel(2, time_str, 0.45, 0.01, 0.1, 0x000000, 0.0, "Arial")
+        self.setLabel(
+            LabelIds.TIME,
+            time_to_string(time),
+            0.45,
+            0.01,
+            0.1,
+            0x000000,
+            0.0,
+            "Arial",
+        )
+
+    def draw_event_messages(self):
+        """Visualize (draw) the event messages from queue"""
+        rev = reversed(self.event_messages_to_draw)
+        for i, (time, msg) in enumerate(rev, start=1):
+            message = "{} - {}".format(time_to_string(time), msg)
+            self.setLabel(
+                getattr(LabelIds, "EVENT_MESSAGE_{}".format(i)),
```

Comment:
@Adman is this actually better than having an `Enum` and doing something like

```python
LableIds[f"EVENT_MESSAGE_{i}"]
```

I honestly don't know, but I guess you'll know :)

Reply:
Now that we are using `\n` we need only one label.

Reply:
Refactored, use `\n` and only one label.